File: backend/app/models/Cottage.js
const mongoose = require('mongoose');

const cottageSchema = mongoose.Schema({
  title: { type: String, required: true },
  description: { type: String, required: true },
  imageUrl: { type: String, required: true },
  price: { type: Number, required: true },
  latitude: { type: Number, required: true },
  longitude: { type: Number, required: true },
});

module.exports = mongoose.model('Cottage', cottageSchema);

File: frontend/lib/features/home/domain/entities/cottage.dart
import 'package:equatable/equatable.dart';

class Cottage extends Equatable {
  final String title;
  final String description;
  final String imageUrl;
  final double price;
  final double latitude;
  final double longitude;

  const Cottage(
      {required this.title,
      required this.description,
      required this.imageUrl,
      required this.price,
      required this.latitude,
      required this.longitude});

  @override
  List<Object?> get props =>
      [title, description, imageUrl, price, latitude, longitude];
}
